### DISCUSSION TITLE
GET_VENDOR_SALES_REPORT error

### DISCUSSION THREAD

**Discussion by pbeeharry2701** [amzn/selling-partner-api-models#3785] [General]
Hello I am using the python-amazon-sp-api library in python to get data from brand-analytics-reports using GET_VENDOR_SALES_REPORT.

But each time the report is being generated it is giving a FATAL error.

I have been using the different parameters 

report_options = {
         "reportPeriod": "DAY",
         "distributorView": "MANUFACTURING",         
         "sellingProgram": "RETAIL"
        # Add any additional report options as needed
    }

Report_type=ReportType.GET_VENDOR_SALES_REPORT
res = Reports(credentials=credentials, marketplace=Marketplaces.DE)
data = res.create_report(reportType=Report_type,reportOptions = report_options)

![image](https://github.com/amzn/selling-partner-api-docs/assets/153748774/371d0536-5cc5-4bec-8696-f63c28339db9)

I have checked the errors description based on the reportdocumentid, it is saying that it needs dataStartTime and dataEndTime must be supplied

![image](https://github.com/amzn/selling-partner-api-docs/assets/153748774/c91dd570-32c4-482b-89a1-5944c777dcf1)

but then when i provide same, it says this report type doesn't require dataStartTime and dataEndTime.

does somoene face same issue and is the solution solution?

Thanks.

**Comment by cookie8monster** [comment]
Wanted to see if you ever were able to solve this issue. I am running into the something very similar with vendor inventory report

**Comment by chrisb-80202** [comment]
I use sp-api to call GET_VENDOR_SALES_REPORT and was receiving a Fatal error in processingStatus. Here's the reply from sp-api support.

I understand you're getting "FATAL" error when requesting GET_VENDOR_SALES_REPORT.

I would like to inform that if you request a report for a reporting period that is not yet available, the report will not be generated and processingStatus for the report will be FATAL. You can get the processing status of the report by polling the getReport operation of the Reports API.

Vendor sales data for a given day is available 168 hours after that day.

For example, data from 2021-12-03 will be available for this report no later than 2021-12-11T00:00:00. If you request a report for a reporting period that is not yet available, the report will not be generated and processingStatus for the report will be FATAL.

Therefore, I request you to make sure give date which is more than 7 days old or 168 hours before. Please check if you're requesting a report for the period which is not yet available and correct the period to get the report.

**Comment by robertdbailey** [comment]
The body of your report request for GET_VENDOR_SALES_REPORT needs to be the following, and note the "dataStartTime" and "dataEndTime" are NOT inside the reportOptions parameter:
```
{
    "reportType": "GET_VENDOR_SALES_REPORT",
    "marketplaceIds": [{{marketplace_id}}],
    "reportOptions": {
        "reportPeriod": "DAY",
        "sellingProgram": "RETAIL",
        "distributorView": "MANUFACTURING"
    },
    "dataStartTime": "2025-10-01",
    "dataEndTime": "2025-10-31"
}
```
...so I would imagine there is an option with the Python library to pass those in addition to your reportOptions object.
